### PR TITLE
Convert deps to Go modules, add BitBox02 dependency for the HSM

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

**Comment by benma** [comment]
I skipped `go mod vendor` for now, which means that all `go build`, `go test` etc. automatically fetch the deps from a cache proxy based on the go.sum file. 

We can add vendoring back later if desired.

**Comment by 0xB10C** [comment]
Thank you for doing this :+1:. Looks good to me and `make ci` runs fine locally.

IMO @Stadicus should give it a try as well to make sure everything is setup correctly for him to build mender images.

File: middleware/src/rpcserver/mocks/Middleware.go [hunk 1]
```diff
@@ -68,6 +66,20 @@ func (_m *Middleware) EnableRootLogin(_a0 rpcmessages.ToggleSettingArgs) rpcmess
 	return r0
 }
 
+// EnableSSHPasswordLogin provides a mock function with given fields: _a0
+func (_m *Middleware) EnableSSHPasswordLogin(_a0 rpcmessages.ToggleSettingArgs) rpcmessages.ErrorResponse {
```
Comment (TheCharlatan) [inline middleware/src/rpcserver/mocks/Middleware.go]: Why is this part of the diff?

Comment (benma) [reply middleware/src/rpcserver/mocks/Middleware.go]: I could swear I took it out :o Fixed with a force push.

Comment (0xB10C) [reply middleware/src/rpcserver/mocks/Middleware.go]: It was never added in https://github.com/digitalbitbox/bitbox-base/pull/237. Apparently `go generate` wasn't there run after renaming the RPCs.

**Comment by benma** [comment]
>  Looks good to me and `make ci` runs fine locally.

 `make ci` is not run by Travis because assembling the image takes too long? Maybe Travis could still build and tests the Go code and potentially other things which are not heavy.

**Comment by TheCharlatan** [comment]
> `make ci` is not run by Travis because assembling the image takes too long? Maybe Travis could still build and tests the Go code and potentially other things which are not heavy.

Which is what we do: https://travis-ci.org/digitalbitbox/bitbox-base/builds/607586093?utm_source=github_status&utm_medium=notification

File: middleware/src/rpcserver/mocks/Middleware.go (hunk 1 above)
Comment (TheCharlatan) [reply middleware/src/rpcserver/mocks/Middleware.go]: Annoying, might make sense to add a check in travis to make sure that go generate does not produce new output.

**Comment by Stadicus** [comment]
> IMO @Stadicus should give it a try as well to make sure everything is setup correctly for him to build mender images.

Will try do build the Go tools as usual using `make docker-build-go`, but there are no direct dependencies from within building the image (just the binaries must be available at `./bin/go/`)

**Comment by Stadicus** [comment]
Command `make docker-build-go` works without any issues, but I did not test the resulting binaries.

**Comment by TheCharlatan** [comment]
Can you rebase?

File: middleware/go.mod [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module github.com/digitalbitbox/bitbox-base/middleware
+
+go 1.13
+
+require (
+	github.com/btcsuite/btcd v0.20.0-beta
```
[the remark this answers is not in the document]
Comment (benma) [reply middleware/go.mod]: Rebased, this line is now gone indeed.

I also upgraded all other deps using `go get -u ./...`.

**Comment by benma** [comment]
I merged now, trusting the two approvals means you tested running the middleware :smile:

**Pull request merged**